components/style/properties/longhand/inherited_text.mako.rs

```diff
@@ -737,7 +737,194 @@ ${helpers.single_keyword("text-align-last",
+                has_fill = true;
+                if fill {
+                    try!(dest.write_str("filled"));
+                } else {
+                    try!(dest.write_str("open"));
+                }
+            }
+            if let Some(shape) = self.shape {
+                if has_fill {
+                    try!(dest.write_str(" "));
+                }
+                try!(shape.to_css(dest));
+            }
+            Ok(())
+        }
+    }
+    impl ToCss for computed_value::Keyword {
+        fn to_css<W>(&self, dest: &mut W) -> fmt::Result where W: fmt::Write {
+            if self.fill {
+                try!(dest.write_str("filled"));
+            } else {
+                try!(dest.write_str("open"));
+            }
+            try!(dest.write_str(" "));
+            self.shape.to_css(dest)
+        }
+    }
+
+    define_css_keyword_enum!(ShapeKeyword:
+                             "dot" => Dot,
+                             "circle" => Circle,
+                             "double-circle" => DoubleCircle,
+                             "triangle" => Triangle,
+                             "sesame" => Sesame);
 
+    impl ShapeKeyword {
+        pub fn char(&self, fill: bool) -> &str {
+            match *self {
+                ShapeKeyword::Dot => if fill { "\u{2022}" } else { "\u{25e6}" },
+                ShapeKeyword::Circle =>  if fill { "\u{25cf}" } else { "\u{25cb}" },
+                ShapeKeyword::DoubleCircle =>  if fill { "\u{25c9}" } else { "\u{25ce}" },
+                ShapeKeyword::Triangle =>  if fill { "\u{25b2}" } else { "\u{25b3}" },
+                ShapeKeyword::Sesame =>  if fill { "\u{fe45}" } else { "\u{fe46}" },
+            }
+        }
+    }
+
+    #[inline]
+    pub fn get_initial_value() -> computed_value::T {
+        computed_value::T::None
+    }
+
+    impl ToComputedValue for SpecifiedValue {
+        type ComputedValue = computed_value::T;
+
+        #[inline]
+        fn to_computed_value(&self, context: &Context) -> computed_value::T {
+            match *self {
+                SpecifiedValue::Keyword(ref keyword) => {
+                    let default_shape = if context.style().get_inheritedbox()
+                                                  .clone_writing_mode() == writing_mode::horizontal_tb {
+                        ShapeKeyword::Circle
+                    } else {
+                        ShapeKeyword::Sesame
+                    };
+                    computed_value::T::Keyword(computed_value::Keyword {
+                        fill: keyword.fill.unwrap_or(true),
+                        shape: keyword.shape.unwrap_or(default_shape)
+                    })
+                },
+                SpecifiedValue::None => computed_value::T::None,
+                SpecifiedValue::String(ref s) => {
+                    let string = if s.len() > 1 {
+                        s.chars().nth(0).unwrap().to_string()
+                    } else { s.clone() };
```

Comment:
Else part is not empty actually, it might be just one letter. And we need to change this line to use graphemes. I filed an easy issue for it, but your suggestion looks cleaner and I will change here and update the issue.